azure-commons-core/src/main/java/com/microsoft/jenkins/azurecommons/core/credentials/TokenCredentialData.java

```diff
@@ -110,11 +116,56 @@ public void setType(int type) {
         this.type = type;
     }
 
+    //
+    // copy from 'org.apache.commons.lang3.SerializationUtils' to avoid the trans-classloader issue
```

Comment:
Apart from this, the only place we used `commons-lang3` is in `MsiTokenCredentials`, for the `StringUtils.isEmpty` call. We can copy that out and get rid of `commons-lang3` dependency completely.

Reply:
Noticed this but I'm not sure whether removing the dependency will affect other plugins since it's used comprehensively.